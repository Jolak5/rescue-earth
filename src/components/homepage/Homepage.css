.forex-container {
  color: black;
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: auto;
}

header {
  display: flex;
  justify-content: space-between;
  padding: 1.5rem;
  background-color: #587be6;
  border-bottom: 1px solid black;
}

header h1 {
  font-weight: bold;
  font-size: 1.5rem;
  color: white;
}

.eachItem {
  background-color: rgb(68, 149, 243);
  padding: 20px 10px;
  display: flex;
  justify-content: space-between;
  color: white;
  border: 1px white solid;
  border-collapse: collapse;
  align-items: flex-start;
}

.eachItem:nth-child(odd) {
  background-color: #587be6;
}

.global {
  background-image: url("../map.jpg");
  padding: 20% 3%;
  background-size: cover;
}

.show-more {
  background-color: transparent;
  text-align: right;
  cursor: pointer;
}

.global-map {
  overflow: hidden;
  opacity: 0.3;
}

.stats {
  text-align: left;
  padding: 10px 10px;
  display: flex;
  flex-direction: column;
}

.eachItem div {
  display: flex;
  flex-direction: column;
  gap: 20px;
}

.stats h1 {
  font-size: 1.5rem;
}

.search-header h1 {
  display: none;
}

.article div h1 {
  color: black;
  font-size: 2.5rem;
}

.article div p {
  color: black;
  font-size: 1rem;
  opacity: 0.9;
}

.stats-countries {
  background-color: rgb(38, 116, 205);
  padding: 10px 10px;
  color: white;
}

svg {
  width: 20px;
  fill: #feaa49;
  cursor: pointer;
}

.search {
  fill: white;
  width: 30px;
}

.search-header {
  align-items: center;
}

form {
  display: flex;
  gap: 5px;
}

input {
  border-radius: 5px;
  border-color: transparent;
}

.submit:hover {
  background-color: #feaa49;
  color: white;
}

@media screen and (min-width: 756px) {
  .forex-container {
    grid-template-columns: 1fr 1fr 1fr 1fr 1fr;
  }

  header {
    padding: 1rem;
  }

  .global {
    background-image: url("../map.jpg");
    padding: 8% 5%;
    background-size: cover;
  }

  .global-map {
    display: none;
  }

  .article div {
    display: flex;
    align-items: center;
    justify-content: space-between;
  }

  .article div h1 {
    font-size: 3rem;
  }

  .article div p {
    font-size: 1.5rem;
    text-align: right;
    color: black;
  }
}
