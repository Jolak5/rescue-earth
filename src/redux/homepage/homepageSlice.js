import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios';

const url = 'https://coronavirus.m.pipedream.net/';

export const getForexData = createAsyncThunk(
  'data/getForexData',
  async () => {
    try {
      const res = await axios.get(url);
      return res.data;
    } catch (error) {
      return error.message;
    }
  },
);

export const ForexSlice = createSlice({
  name: 'Forex',
  initialState: {
    Forex: [],
    Global: [],
    searchFilter: [],
    forexItem: [],
    isLoading: true,
  },
  reducers: {
    // display individual item details
    renderItem: (state, action) => {
      const id = action.payload;
      const forexFiltered = state.Forex.filter((Forex) => Forex.id === id);

      return { ...state, forexItem: forexFiltered };
    },
    // filter out items with the search
    filterItem: (state, action) => {
      const value = action.payload.toLowerCase();
      // eslint-disable-next-line max-len
      const filteredItem = state.Forex.filter((forex) => ((forex.country.toLowerCase()).includes(value)) === true);
      return { ...state, searchFilter: filteredItem, isLoading: false };
    },
    clearFilteredState: (state) => ({ ...state, forexItem: [] }),
  },
  extraReducers: (builders) => {
    builders
      .addCase(getForexData.pending, (state) => ({
        ...state,
        isLoading: true,
      }))
      .addCase(getForexData.fulfilled, (state, action) => {
        const data = action.payload;
        const covidData = data.rawData.map((item) => ({
          id: item.Combined_Key,
          country: item.Country_Region,
          confirmed: item.Confirmed,
          Deaths: item.Deaths,
          Fatality: item.Case_Fatality_Ratio,
          Last_Update: item.Last_Update,
          Incident_rate: item.Incident_Rate,
          active: item.Active,
        }));
        return {
          ...state, isLoading: false, Forex: covidData, Global: [action.payload.summaryStats.global],
        };
      })
      .addCase(getForexData.rejected, (state) => ({
        ...state,
        isLoading: false,
      }));
  },
});
export const { renderItem, filterItem, clearFilteredState } = ForexSlice.actions;
export default ForexSlice.reducer;
