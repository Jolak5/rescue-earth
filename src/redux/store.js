import { configureStore } from '@reduxjs/toolkit';
import pollutionReducer from './homepage/homepageSlice';

export default configureStore({
  reducer: {
    pollution: pollutionReducer,
  },
});
