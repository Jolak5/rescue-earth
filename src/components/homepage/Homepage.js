import React from 'react';
import Button from 'react-bootstrap/Button';
import Container from 'react-bootstrap/Container';
import Col from 'react-bootstrap/Col';
import './Homepage.css';
import { useDispatch, useSelector } from 'react-redux';
import { renderItem } from '../../redux/homepage/homepageSlice';
import { openModal } from '../../redux/modal/modalSlice';

export default function Homepage() {
  const dispatch = useDispatch();
  const covid = useSelector((state) => state.Forex.Forex);
  const newCovid = covid.slice(0, 10);
  return (
    <Container className="forex-container">
      {newCovid.map((item) => (
        <Col className="eachItem" key={item.id}>
          <h2>{item.country}</h2>
          <p>{item.death}</p>
          <Button onClick={() => {
            dispatch(renderItem(item.id));
            dispatch(openModal());
          }}
          >
            Kayode

          </Button>
        </Col>
      ))}
    </Container>
  );
}
