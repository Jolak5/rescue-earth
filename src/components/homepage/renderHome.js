import React from 'react';
import './Homepage.css';
import { useDispatch, useSelector } from 'react-redux';
import { NavLink, useParams } from 'react-router-dom';
import { renderItem } from '../../redux/homepage/homepageSlice';
import Icon, { Search } from '../icon';

export default function RenderHome() {
  const dispatch = useDispatch();
  const covid = useSelector((state) => state.Forex.Forex);
  const global = useSelector((state) => state.Forex.Global);
  const { countryId } = useParams();
  const newCovid = covid.slice(0, 20);
  return (
    <div>
      <header className="header">
        <h1>CoviLab </h1>
        <NavLink to="search"><Search /></NavLink>

      </header>
      <div className="global">

        <div className="global-details stats">
          {global.map((item) => (
            <article className="article" key={item.id}>
              <div>
                <h1>
                  Global
                  {' '}
                  <br />
                  {' '}
                  Covid-19

                  Stats
                </h1>
                <p>
                  Confirmed cases:
                  {' '}
                  {item.confirmed}
                  <br />
                  Total deaths:
                  {' '}
                  {item.deaths}

                </p>
              </div>
            </article>
          ))}
        </div>
      </div>
      <div className="stats-countries">STATS BY COUNTRY</div>
      <div className="forex-container">

        {newCovid.map((item) => (
          <article className="eachItem" key={item.Deaths}>

            <div>
              <h2>{item.country}</h2>
              <p>
                {item.Deaths}
                {' '}
                deaths
              </p>
            </div>
            <NavLink to={countryId}>
              <button
                className="show-more"
                type="button"
                onClick={() => {
                  dispatch(renderItem(item.id));
                }}
              >
                <Icon />
              </button>
            </NavLink>
          </article>
        ))}
      </div>
    </div>
  );
}
