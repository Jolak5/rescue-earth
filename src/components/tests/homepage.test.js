import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import { cleanup, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Homepage from '../homepage/Homepage';
import { renderItem, filterItem } from '../../redux/homepage/homepageSlice';
import store from '../../redux/store';

afterEach(cleanup);

describe('Homepage', () => {
  it('Homepage to match Snapshot', () => {
    const quotes = render(
      <Router>
        <Provider store={store}>
          <Homepage />
        </Provider>
      </Router>,
    );
    expect(quotes).toMatchSnapshot();
  });
});

describe('renderItem reducers', () => {
  it('should render items', async () => {
    await store.dispatch(renderItem('Algeria'));
    expect(store.getState().Forex.forexItem).toBeDefined();
  });
  it('should filter Item', () => {
    const filtered = store.dispatch(filterItem('Nigeria'));
    expect(filtered).toBeDefined();
  });
});
