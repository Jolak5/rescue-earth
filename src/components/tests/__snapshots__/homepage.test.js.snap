// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage Homepage to match Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div>
          <header
            class="header"
          >
            <h1>
              CoviLab 
            </h1>
            <a
              class=""
              href="/search"
            >
              <svg
                class="search"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
                />
              </svg>
            </a>
          </header>
          <div
            class="global"
          >
            <div
              class="global-details stats"
            />
          </div>
          <div
            class="stats-countries"
          >
            STATS BY COUNTRY
          </div>
          <div
            class="forex-container"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div>
        <header
          class="header"
        >
          <h1>
            CoviLab 
          </h1>
          <a
            class=""
            href="/search"
          >
            <svg
              class="search"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
              />
            </svg>
          </a>
        </header>
        <div
          class="global"
        >
          <div
            class="global-details stats"
          />
        </div>
        <div
          class="stats-countries"
        >
          STATS BY COUNTRY
        </div>
        <div
          class="forex-container"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
